allure-testng/src/test/java/io/qameta/allure/testng/AllureTestNgTest.java

```diff
@@ -109,6 +110,28 @@ public static Object[][] parallelConfiguration() {
         };
     }
 
+    @AllureFeatures.Fixtures
+    @Issue("356")
+    @Test
+    public void shouldUnsetLogConfigFailProperty() {
+        AllureTestNgConfig allureTestNgConfig = AllureTestNgConfig.loadConfigProperties();
+        assertThat(allureTestNgConfig.isLogConfigurationFailures()).isTrue();
+        allureTestNgConfig.setLogConfigurationFailures(false);
+        assertThat(allureTestNgConfig.isLogConfigurationFailures()).isFalse();
+    }
+
+    @AllureFeatures.Fixtures
+    @Issue("356")
+    @Test
+    public void shouldNotDisplayConfigurationFailsAsTests() {
+        System.setProperty(ALLURE_TESTNG_LOG_CONFIGURATION_FAILURES, "false");
```

Comment:
Please, use the config object instead. This test has side effects (and the system property won't be cleared in case of test failure).

I suggest you create a package-private constructor of AllureTestNg:

```java
AllureTestNg(final AllureLifecycle lifecycle,
                       final AllureTestNgTestFilter testFilter,
                       final AllureTestNgConfig config) {
    this.lifecycle = lifecycle;
    this.testFilter = testFilter;
    this.config = config;
}
```

Then add a method overload for `runTestNgSuites` that accepts `AllureTestNgConfig` as a second parameter.

Reply:
Ive created a package-private constructor as suggested, and applied overloads for the runTestNgSuites, as well as reordered them so that they are all near one another for easy cross-reference.

I have also removed the same setup from the other configuration test which tests Hiding Disabled Tests to ensure that it also wouldn't have unintended side effects or an uncleared system property.